src/main/java/seedu/address/model/appointment/Appointment.java

```diff
@@ -24,56 +18,78 @@ public class Appointment {
         this.description = description;
     }
 
+    /**
+     * Retrieves the date and time of the appointment.
+     *
+     * @return The date and time of the appointment.
+     */
     public DateTime getDateTime() {
         return dateTime;
     }
 
+    /**
+     * Sets the date and time of the appointment.
+     *
+     * @param dateTime The new date and time for the appointment.
+     */
+    public void setDateTime(DateTime dateTime) {
+        this.dateTime = dateTime;
+    }
+
+    /**
+     * Retrieves the student associated with the appointment.
+     *
+     * @return The student associated with the appointment.
+     */
     public Person getStudent() {
         return student;
     }
 
+    /**
+     * Retrieves the description associated with the appointment.
+     *
+     * @return The description associated with the appointment.
+     */
     public Description getDescription() {
         return description;
     }
 
     /**
-     * Returns true if both appointments have the same fields.
+     * Sets the student associated with the appointment.
+     *
+     * @param student The new student for the appointment.
      */
-    @Override
-    public boolean equals(Object other) {
-        if (other == this) {
-            return true;
-        }
-
-        // instanceof handles nulls
-        if (!(other instanceof Appointment)) {
-            return false;
-        }
-
-        Appointment otherAppointment = (Appointment) other;
-        return student.equals(otherAppointment.student)
-                && description.equals(otherAppointment.description)
-                && dateTime.equals(otherAppointment.dateTime);
+    public void setStudent(Person student) {
+        this.student = student;
     }
 
-    @Override
-    public int hashCode() {
-        // use this method for custom fields hashing instead of implementing your own
-        return Objects.hash(student, description, dateTime);
+    /**
+     * Sets the description associated with the appointment.
+     *
+     * @param description The new description for the appointment.
+     */
+    public void setDescription(Description description) {
+        this.description = description;
     }
 
-    @Override
-    public String toString() {
-        return new ToStringBuilder(this)
-                .add("student", student)
-                .add("description", description)
-                .add("dateTime", dateTime)
-                .toString();
+    /**
+     * Checks if this appointment is the same as another appointment.
+     * Two appointments are considered the same if they have the same description.
+     *
+     * @param otherAppointment The appointment to compare with.
+     * @return {@code true} if the appointments are the same, {@code false} otherwise.
+     */
+    public boolean isSameAppointment(Appointment otherAppointment) {
+        if (otherAppointment == this) {
+            return true;
+        }
+
+        return otherAppointment != null
```

Comment:
I believe if we want to have some form of checks between appointments, it will be the checking of whether they are any overlap in time period for appointments (Since a counsellor can only handle 1 appointment at one time)